platform/android/app/src/main/kotlin/coop/polypoly/polypod/MainActivity.kt

```diff
@@ -4,16 +4,32 @@ import android.app.AlertDialog
 import android.content.Intent
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import coop.polypoly.core.Core
 import coop.polypoly.polypod.core.UpdateNotification
 import coop.polypoly.polypod.features.FeatureStorage
+import coop.polypoly.polypod.logging.LoggerFactory
 
 class MainActivity : AppCompatActivity() {
+    companion object {
+        @Suppress("JAVA_CLASS_ON_COMPANION")
+        private val logger = LoggerFactory.getLogger(javaClass.enclosingClass)
+    }
 
     private var onboardingShown = false
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        val language = Language.determine(this@MainActivity)
+        val bootstrapResult = Core.bootstrapCore(language)
+        if (bootstrapResult.isSuccess) {
+            logger.info("Core is bootstraped!")
+        } else {
+            logger.error(
+                "Failed to boostrap core",
+                bootstrapResult.exceptionOrNull()
+            )
```

Comment:
I feel like we're asking for trouble here - you don't actually quit the app, so we're just going to have undefined behaviour at runtime? Seems like in this case the app would crash when trying to show the feature list. That's gonna be nasty to debug...

How about we skip any logging and just go for:

```
if (!bootstrapResult.isSuccess) {
    throw new RuntimeException("Failed to bootstrap core: ${bootstrapResult.exceptionOrNull}")
}
```

? That way, we deliberately crash the app with a specific message, which should make it a lot easier to identify in crash reports.

Reply:
True, it will be safer to crash the app.